Client/Application/Delegates/AppDelegate.swift

```diff
@@ -295,46 +295,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
                 Preferences.URP.referralLookupOutstanding.value = isFirstLaunch
             }
             
-            if Preferences.URP.referralLookupOutstanding.value == true {
-                var refCode: String?
-                
-                let savedRefCode: String? = ProcessInfo().operatingSystemVersion.majorVersion > 13
-                    ? nil : UIPasteboard.general.string
-                
-                if Preferences.URP.referralCode.value == nil {
-                    UrpLog.log("No ref code exists on launch, attempting clipboard retrieval")
-                    refCode = UserReferralProgram.sanitize(input: savedRefCode)
-                }
-                
-                if refCode != nil {
-                    UrpLog.log("Clipboard ref code found: " + (savedRefCode ?? "!Clipboard Empty!"))
-                    UrpLog.log("Clearing clipboard.")
-                    UIPasteboard.general.clearPasteboard()
-                }
-                
-                urp.referralLookup(refCode: refCode) { referralCode, offerUrl in
-                    // Attempting to send ping after first urp lookup.
-                    // This way we can grab the referral code if it exists, see issue #2586.
-                    self.dau.sendPingToServer()
-                    if let code = referralCode {
-                        let retryTime = AppConstants.buildChannel.isPublic ? 1.days : 10.minutes
-                        let retryDeadline = Date() + retryTime
-                        
-                        Preferences.NewTabPage.superReferrerThemeRetryDeadline.value = retryDeadline
-                        
-                        self.browserViewController.backgroundDataSource
-                            .fetchSpecificResource(.superReferral(code: code))
-                    } else {
-                        self.browserViewController.backgroundDataSource.startFetching()
-                    }
-                    
-                    guard let url = offerUrl?.asURL else { return }
-                    self.browserViewController.openReferralLink(url: url)
-                }
-            } else {
-                urp.pingIfEnoughTimePassed()
-                browserViewController.backgroundDataSource.startFetching()
-            }
+            handleReferralLookup(urp, checkClipboard: false)
```

Comment:
@iccub - I'm not sure if I understand this call. If `checkClipboard` is false, `referral code` won't be accessed right?

Reply:
Yes, if it is false we will try to do a IP-matching without accessing clipboard

Reply:
That makes sense, but I'm confused about why this is needed if we already have this: https://github.com/brave/brave-ios/pull/2837/files#diff-d7437916ab14b41a34d7eb8e2a8d3afdR40

Reply:
@jumde  The privacy consent screen(`OnboardingPrivacyConsentViewController`) will not show if user has nothing in the clipboard, 
for this case we handle referral lookup here, and we force it to not access clipboard
so there's 3 ways to handle the referral now

1. No clipboard - do IP check here
2. Clipboard present - user consents - do clipboard referral lookup
3. Clipboard present - user denies - do IP referral lookup